x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/Foldables.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.ql.expression;
+
+import org.elasticsearch.xpack.ql.QlIllegalArgumentException;
+import org.elasticsearch.xpack.ql.type.DataType;
+import org.elasticsearch.xpack.ql.type.DataTypeConverter;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public abstract class Foldables {
+    
+    @SuppressWarnings("unchecked")
+    private static <T> T valueOf(Expression e, DataType to) {
```

Comment:
I would remove this from ql since it's a trappy class. It uses `DataTypeConverter` which is QL specific and SQL would end up using that instead of `SqlDataTypeConverter`.
That is, `Foldables` is really reusable because of that and as such, I would just inline the methods.

Reply:
+1